client/my-sites/site-settings/jetpack-credentials/credentials-setup-flow/index.jsx

```diff
@@ -0,0 +1,139 @@
+	}
+
+	reset = () => {
+		this.setState( { currentStep: 'start' } );
+	};
+
+	goToNextStep = () => {
+		switch ( this.state.currentStep ) {
+			case 'start':
+				this.setState( { currentStep: 'tos' } );
+				return;
+			case 'tos':
+				this.setState( { currentStep: 'form' } );
+				return;
+		}
+	};
+
+	autoConfigure = () => this.props.autoConfigCredentials( this.props.siteId );
+
+	renderStepStart() {
+		const {
+			translate
+		} = this.props;
+
+		return (
+			<CompactCard
+				className="credentials-setup-flow__setup-start"
+				onClick={ this.goToNextStep }
+			>
+				<Gridicon icon="add-outline" size={ 48 } className="credentials-setup-flow__header-gridicon" />
+				<div className="credentials-setup-flow__header-text">
+					<h3 className="credentials-setup-flow__header-text-title">{ translate( 'Add site credentials' ) }</h3>
+					<h4 className="credentials-setup-flow__header-text-description">
+						{ translate( 'Used to perform automatic actions on your server including backing up and restoring.' ) }
+					</h4>
+				</div>
+			</CompactCard>
+		);
+	}
+
+	renderStepTos() {
+		const {
+			isPressable,
+			translate
+		} = this.props;
+
+		return (
+			<CompactCard
+				className="credentials-setup-flow__tos"
+				highlight="info"
+			>
+				<Gridicon icon="info" size={ 48 } className="credentials-setup-flow__tos-gridicon" />
+				<div className="credentials-setup-flow__tos-text">
+					{
+						isPressable
+							? translate( 'WordPress.com can obtain the credentials from your ' +
+							'current host which are necessary to perform site backups and ' +
+							'restores. Do you want to give WordPress.com access to your ' +
+							'host\'s server?' )
+							: translate( 'By adding your site credentials, you are giving ' +
+							'WordPress.com access to perform automatic actions on your ' +
+							'server including backing up your site, restoring your site, ' +
+							'as well as manually accessing your site in case of an emergency.' )
+					}
+				</div>
+				{
+					isPressable
+						? <Button primary onClick={ this.autoConfigure }>{ translate( 'Auto Configure' ) }</Button>
+						: <Button primary onClick={ this.goToNextStep }>{ translate( 'Ok, I understand' ) }</Button>
+				}
+			</CompactCard>
+		);
+	}
+
+	renderStepForm() {
+		const {
+			formIsSubmitting,
+			updateCredentials,
+			siteId
+		} = this.props;
+
+		return (
+			<CompactCard>
+				<CredentialsForm { ...{
+					formIsSubmitting,
+					protocol: 'ssh',
+					host: '',
+					port: '',
+					user: '',
+					pass: '',
+					abspath: '',
+					kpub: '',
+					onCancel: this.reset,
+					siteId,
+					updateCredentials,
+				} } />
+			</CompactCard>
+		);
+	}
+
+	render() {
+		return (
+			<div className="credentials-setup-flow">
+				{ 'start' === this.state.currentStep && this.renderStepStart() }
+				{ 'tos' === this.state.currentStep && this.renderStepTos() }
+				{ 'form' === this.state.currentStep && this.renderStepForm() }
```

Comment:
Cool! This is a much nicer place to be than when we started. We're even ready to extract these into separate React components.

Reply:
I think we can worry about additional react components after we get this merged. Thoughts?

Reply:
It's up to you, though I would ask probably "why not" here since the steps are already extracted into their own functions. In general we like to _either_ inline JSX inside of `render()` _or_ split them into separate components.

Here, we're _adding new code_ so the cost of doing it one way or the other is basically free in that we're not worried about introducing regressions to existing code. If we want to do it one way, let's just do it that way the first time because one of two things will happen: we'll never get back around to it and it will live on forever how we don't want it to; or we'll end up spending more time overall on it because when we come back to update it later (or _deal with it_) we won't be as efficient as we are now (when it's new and when we've been in it so much).

Now, you are free to decide that you don't want to do it the way I'm suggesting, and I will be fine with that. But, what I hope we don't do is push out new code that we think needs immediate refactoring - that's what PRs are for 😉